tests/Tests.AzureAppConfiguration/MapTests.cs

```diff
@@ -0,0 +1,325 @@
+﻿using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+using Azure.Data.AppConfiguration;
+using Azure;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.AzureKeyVault;
+using System.Linq;
+using Azure.Core.Testing;
+using Moq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManagement;
+
+namespace Tests.AzureAppConfiguration
+{
+    public class MapTests
+    {
+        List<ConfigurationSetting> _kvCollection = new List<ConfigurationSetting>
+        {
+            ConfigurationModelFactory.ConfigurationSetting(
+                key: "TestKey1",
+                label: "label",
+                value: "TestValue1",
+                eTag: new ETag("0a76e3d7-7ec1-4e37-883c-9ea6d0d89e63"),
+                contentType: "text"),
+
+            ConfigurationModelFactory.ConfigurationSetting(
+                key: "TestKey2",
+                label: "label",
+                value: "TestValue2",
+                eTag: new ETag("31c38369-831f-4bf1-b9ad-79db56c8b989"),
+                contentType: "text")
+        };
+
+        ConfigurationSetting FirstKeyValue => _kvCollection.First();
+        ConfigurationSetting sentinelKv = new ConfigurationSetting("SentinelKey", "SentinelValue");
+
+        TimeSpan CacheExpirationTime = TimeSpan.FromSeconds(1);
+
+        string _certValue = "Certificate Value from KeyVault";
+        Uri vaultUri = new Uri("https://keyvault-theclassics.vault.azure.net");
+
+        [Fact]
+        public void MapTransformKeyValue()
+        {
```

Comment:
Can you add tests where Map is changing the key/label? What is the expected behavior in both cases? What is the expected behavior if key modifications lead to duplicate keys in IConfiguration?